### ISSUE TITLE
DEFORMABLE DETR Implementation

### ISSUE THREAD

**Issue by thibo73800** [Visual-Behavior/detr-tensorflow#9]
Implement this paper

https://openreview.net/pdf?id=gZ9hCDWe6ke

```
DETR has been recently proposed to eliminate the need for many hand-designed
components in object detection while demonstrating good performance. However,
it suffers from slow convergence and limited feature spatial resolution, due to the
limitation of Transformer attention modules in processing image feature maps. To
mitigate these issues, we proposed Deformable DETR, whose attention modules
only attend to a small set of key sampling points around a reference. Deformable
DETR can achieve better performance than DETR (especially on small objects)
with 10× less training epochs. Extensive experiments on the COCO benchmark
demonstrate the effectiveness of our approach. Code shall be released.
```

**Comment by PhanTask** [comment]
Excuse me, but when will this be implemented? Do you have a specific plan now?

**Comment by thibo73800** [comment]
Yes, we have a working version. I will move everything including mask & deformable and multi-scale training starting by the end of this week.

**Comment by PhanTask** [comment]
@thibo73800 Wow that would be fabulous! Thanks for your great efforts!

**Comment by PhanTask** [comment]
Hi Thibo @thibo73800 , just wondering when will the deformable version be updated. If you still need more time to wrap up things, could you share with me a copy of the code so that I can have a chance to test it? I am currently working on re-implementing TrackFormer, which is based on deformable DETR. Thanks!

**Comment by thibo73800** [comment]
Hi ! I have been quite busy during the last two last weeks with other projects. But the multi-scale training part is almost ready. Once done I will import Deformable. Getting a first working script should not take too long. 

Great for your implementation of trackformer, we did it as well. 

Since you'll probably benefit from getting a WIP version in this repo, as soon as I have a first working script on this repo I will push the code on another branch for you to test it. Probably/Hoping before next Tuesday.

Best,
Thibault

**Comment by PhanTask** [comment]
@thibo73800 that sounds great!! Thanks for your great efforts! You did a great job!

**Comment by thibo73800** [comment]
WIP here https://github.com/Visual-Behavior/detr-tensorflow/pull/25

**Comment by PhanTask** [comment]
@thibo73800 Awesome! Gonna check this out. Thanks!

**Comment by amirbar** [comment]
Thanks for working on this @thibo73800. Do you happen to have some MS COCO results with the Deformable DETR impl? are you able to reproduce similar results as in the paper?
